examples/starwars/modules/datasource/types.bal

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public enum EpisodeEnum {
+    NEWHOPE,
+    EMPIRE,
+    JEDI
+}
+
+public type HumanRecord readonly & record {|
+    readonly string id;
+    string name;
+    string homePlanet?;
+    float height?;
+    int mass?;
+    EpisodeEnum[] appearsIn;
+|};
+
+public type DroidRecord readonly & record {|
+    readonly string id;
+    string name;
+    EpisodeEnum[] appearsIn;
+    string primaryFunction?;
+|};
+
+public type StarshipRecord readonly & record {|
+    readonly string id;
+    string name;
+    float length?;
+    float[][] cordinates?;
+|};
+
+public type ReviewRecord readonly & record {|
+    EpisodeEnum episode;
+    int stars;
+    string commentary?;
+|};
+
+public type FriendsEdgeRecord readonly & record {|
+    readonly string characterId;
+    readonly string friendId;
+|};
```

Comment:
If the record is `readonly` do we need to mark fields `readonly` ? @MaryamZi

Reply:
We can remove `readonly` from the fields.

```ballerina
readonly & record {|
    T a;
|};
```

is effectively 
```ballerina
readonly & record {|
    readonly T a;
|};
```

which is

```ballerina
readonly & record {|
    readonly (readonly & T) a;
|};
```